tests/test_cli_options.py

```diff
@@ -0,0 +1,507 @@
+                "model_name,gpu_uuid,batch_size,concurrency,model_config_path,instance_group,satisfies_constraints,gpu_used_memory,gpu_utilization,gpu_power_usage"),
+            OptionStruct("stringlist", "analyze", "--server-output-fields", None, "a, b, c",
+                "model_name,gpu_uuid,gpu_used_memory,gpu_utilization,gpu_power_usage"),
+
+            # No OP Options:
+            # Option format:
+            # (noop, any MA step, long_flag,)
+            # These commands arent tested directly but are here to ensure that
+            # the count is correct for all options in the config.
+            # Some of these are required to run the subcommand
+            # Others are yaml only options
+            OptionStruct("noop", "profile", "--model-repository"),
+            OptionStruct("noop", "profile", "--profile-models"),
+            OptionStruct("noop", "analyze", "--analysis-models"),
+            OptionStruct("noop", "report", "--report-model-configs"),
+            OptionStruct("noop", "report", "--output-formats", "-o", ["pdf", "csv", "png"], "pdf", "SHOULD_FAIL"),
+            OptionStruct("noop", "yaml_profile", "constraints"),
+            OptionStruct("noop", "yaml_profile", "objectives"),
+            OptionStruct("noop", "yaml_profile", "triton_server_flags"),
+            OptionStruct("noop", "yaml_profile", "perf_analyzer_flags"),
+            OptionStruct("noop", "yaml_profile", "triton_docker_labels"),
+            OptionStruct("noop", "yaml_profile", "triton_server_environment"),
+            OptionStruct("noop", "yaml_analyze", "constraints"),
+            OptionStruct("noop", "yaml_analyze", "objectives"),
+            OptionStruct("noop", "yaml_analyze", "plots"),
+        ]
+        #yapf: enable
+
+        all_tested_options_set = set()
+
+        for option in options:
+            all_tested_options_set.add(self._convert_flag(option.long_flag))
+
+            if option.type in ["bool"]:
+                self._test_boolean_option(option)
+            elif option.type in ["int", "float"]:
+                self._test_numeric_option(option)
+            elif option.type in ["string"]:
+                self._test_string_option(option)
+            elif option.type in ["intlist", "stringlist"]:
+                self._test_list_option(option)
+
+        self._verify_all_options_tested(all_tested_options_set)
+
+    def _verify_all_options_tested(self, all_tested_options_set):
+        cli_option_set = set()
+
+        # Get all of the options in the CLI Configs
+        structs = [
+            CLIConfigProfileStruct, CLIConfigAnalyzeStruct,
+            CLIConfigReportStruct
+        ]
+        for struct in structs:
+            cli = struct()
+            _, config = cli.parse()
+            for key in config.get_config().keys():
+                cli_option_set.add(key)
+
+        self.assertEqual(cli_option_set, all_tested_options_set)
```

Comment:
people are going to hit this assert when adding a new option and not understand why or how to fix it.  It looks like assertEqual takes an optional 3rd arg for the message to print when failing. We should really be utilizing that more in our tests.